### PR TITLE
add jobs definition for upgrade b/w 4.12 stable build

### PR DESCRIPTION
**Pull request by rioliu-rh**
recrate PR for #40947

### CODE REVIEW

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-stable-4.12-upgrade-from-stable-4.12-vsphere-ipi-proxy-fips-cert-rotation-p2-f1

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-stable-4.12-upgrade-from-stable-4.12-azure-aro-cert-rotation-p1-f1

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-stable-4.12-upgrade-from-stable-4.12-gcp-ipi-ovn-ipsec-fips-cert-rotation-p2-f14

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-stable-4.12-upgrade-from-stable-4.12-aws-ipi-ovn-fips-cert-rotation-p2-f1

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-stable-4.12-upgrade-from-stable-4.12-baremetalds-ipi-ovn-ipv4-fips-cert-rotation-p2-f14

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-stable-4.12-upgrade-from-stable-4.12.yaml [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    env:
+      FIPS_ENABLED: "true"
+      RT_ENABLED: "true"
+      UPGRADE_CHECK_RUN_TAGS: '@upgrade-check and @4.12 and @gcp-ipi'
+      UPGRADE_PRE_RUN_TAGS: '@upgrade-prepare and @4.12 and @gcp-ipi'
+    test:
+    - ref: cucushift-installer-check
+    - chain: openshift-upgrade-qe-test-cert-rotation
+    workflow: cucushift-installer-rehearse-gcp-ipi-cco-manual-workload-identity
+- as: gcp-ipi-workload-identity-fips-p2-f1
```
Comment (jianli-wei) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-stable-4.12-upgrade-from-stable-4.12.yaml:111]: I guess `gcp-ipi-workload-identity-fips-p2-f1` had been covered by  the above`gcp-ipi-workload-identity-rt-fips-cert-rotation-p2-f1`. If so, suggest to remove `gcp-ipi-workload-identity-fips-p2-f1`.

BTW, shall the file be named as `openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-stable-4.12.yaml` instead? Sorry I don't quite understand upgrading from stable-4.12 to stable-4.12, thanks!

File: ci-operator/step-registry/openshift/upgrade/qe/test/cert-rotation/openshift-upgrade-qe-test-cert-rotation-commands.sh [hunk 2]
```diff
@@ -126,6 +127,8 @@ oc adm wait-for-stable-cluster
 
 # create new admin.kubeconfig
 oc config new-admin-kubeconfig > "${SHARED_DIR}/admin.kubeconfig"
+oc --kubeconfig="${SHARED_DIR}/admin.kubeconfig" whoami || true
+sleep 2
```
Comment (jianli-wei) [inline ci-operator/step-registry/openshift/upgrade/qe/test/cert-rotation/openshift-upgrade-qe-test-cert-rotation-commands.sh:131]: By the sleeping, does it mean `new-admin-kubeconfig` would generally take some time, so better to sleep a bit before using the new kubeconfig? Thanks!

**Review by jianli-wei — CHANGES_REQUESTED** [review]

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-stable-4.12-upgrade-from-stable-4.12.yaml (hunk 1 above)
Comment (rioliu-rh) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-stable-4.12-upgrade-from-stable-4.12.yaml:111]: sorry for making you confuse. this PR is just used to trigger regression test. won't be merged.

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-stable-4.12-upgrade-from-stable-4.12-vsphere-ipi-proxy-fips-cert-rotation-p2-f1

**Pull request closed without merging**